pyOCD/gdbserver/context_facade.py

```diff
@@ -117,12 +117,16 @@ def getSignalValue(self):
         if self._context.core.isDebugTrap():
             return signals.SIGTRAP
 
-        fault = self._context.readCoreRegister('xpsr') & 0xff
-        try:
-            signal = FAULT[fault]
-        except:
-            # If not a fault then default to SIGSTOP
-            signal = signals.SIGSTOP
+        # If not a fault then default to SIGSTOP
+        signal = signals.SIGSTOP
+        
+        if self._context.core.isVectorCatch():
+            fault = self._context.core.readCoreRegister('xpsr') & 0xff
```

Comment:
I inserted the `.core` here because without, we were getting the thread xpsr value (with IPSR=0), when we're interested in the live xpsr value, if trying to establish why a thread faulted. I wanted the SIGSEGV to be associated with the source thread, not the Handler mode.

But ".core" bypasses both the OS awareness and the cache, right? Would `self._context.core.getTargetContext().readCoreRegister` make sense?

Should `isVectorCatch` internally use `_target_context` to do its memory reads, to cache, rather than doing them raw?